source/components/AvatarStyled/index.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+import Badge from './Badge';
+import DefaultAvatar from './DefaultAvatar';
+import UserAvatar from './UserAvatar';
+
+function getBorderWidth(size) {
+    const diameter = parseInt(size, 10);
+    let borderWidth = 2;
+    if (diameter >= 48) {
+        borderWidth = 3;
+    }
+    if (diameter >= 120) {
+        borderWidth = 4;
+    }
+    return `${borderWidth}px`;
+}
+
+function getBadgeSize(size) {
+    return (0.2539 * parseInt(size, 10)) + 8;
+}
+
+const Link = styled.a`
+    color: #000;
+    height: 100%;
+    width: 100%;
+
+    &:hover {
+        color: #00acac;
+    }
+`;
+
+const Wrapper = styled.div`
+    display: inline-block;
+    height: ${props => `${props.size}px`};
+    min-width: ${props => `${props.size}px`};
+    position: relative;
+    width: ${props => `${props.size}px`};
+
+    ${Link} ${DefaultAvatar}:hover {
+        fill: ${props => (props.href ? '#00acac' : '#bed1cf')};
+    }
+
+    ${Link} ${UserAvatar}:hover {
+        border-color: ${props => (props.href ? '#00acac' : '#bed1cf')};
+    }
+`;
+
+/**
+ * A `styled-component` version of Avatar component
+ */
+class AvatarStyled extends React.Component {
```

Comment:
Can we switch this to a functional component? For component did mount you can use `useEffect(()=>{// mount stuff here}, [])`